### ISSUE TITLE
Feature: Being able to trigger smartlock without starting the UIs

### ISSUE THREAD

**Issue by Nexcius** [schibsted/account-sdk-android#95]
From: Lasse Magnussen

**Comment by Astach** [comment]
Launch smartock without the UI isn't possible. Smarlock uses intent  catch by the activity in `onActivityResult`. The SDK activity intercepts the result and depending on the result :
- Credentials found: Inject credentials in the login flow, and start the login flow as usual ( that means all checks about terms, required fields etc. are still done.)
- Only Identifier found: Smarlock injects identifier to the login flow and start the usual login flow with the identifier prefilled.
- Any failure: provide you back an intent to restart locking flow with smarlock disabled.

If we provide smartlock without UI that means:
- The client has to implement the login contract himself
- When credentials are found the client has to handle it himself and pass credentials to the controller implementation the client has made.
- If the identifier only is provided by smartlock the client has to start the login activity anyway, and provides identifiers to the UI in order to prefill the identifier screen.

Basically, it means the UI  is useless since the client has to implement contracts and all the logic himself.

Also starting smartlock without cliking on a button is doable, but if it the smartlock can't silently login the user, the SDK will fallback to the default UI login flow.
**What we can do** is provide a way to the client to say "I want to fallback to the default UI login flow if smartlock can't silently login" then
- if the client say yes : we fallback to the normal login flow with prefilled identifier
- if the client say no : smartlock will fail and no further action will be done.

**Comment by lassem** [comment]
You could leverage a headless fragment in order to catch the `onActivityResult` without actually opening the activity. It is not super-awesome, but it is usually better than delegating the work to the client. For that to work the client would have to call a setup function on startup. You would need access to a `FragmentManager` somehow, either by the client providing an activity or by cheating and adding an `ActivityLifecycleCallbacks` hook on the application object.

What we would like is something like what is specified in the bottom om @Astach's comment above:
* Ask Smartlock to log us in.
* If Smartlock has no credentials or login fails; fail silently and allow nature to take its course; thereby allowing the user to login through regular means. Usually this involves clicking a login button somewhere.

I could provide assistance if you want me to.

**Comment by Astach** [comment]
#114
